### PR TITLE
Core 1444 - Allow users to refresh cache of their publicly linked files

### PR DESCRIPTION
**Pull request by sri2k1us**
image

image

### CODE REVIEW

File: public/static/locales/en/data.json [hunk 1]
```diff
@@ -122,6 +122,10 @@
     "requestDOIPrompt": "Please read the <manual>DOI Manual</manual> and check if you meet the requirements for submitting a DOI request.",
     "requestDOIAgreement": "By clicking <b>I need a DOI</b>, I have read and agree to the terms of the <agreement> Data Commons </agreement>.",
     "refresh": "Refresh",
+    "refreshCache": "Refresh Cache",
+    "refreshCacheTooltip": "Public file(s) are cached for performance. Select `Refresh Cache` to update the cache.",
```
Comment (psarando) [inline public/static/locales/en/data.json:126]:
```suggestion
    "refreshCacheTooltip": "Public files are cached for performance. Select `Refresh Cache` to update the cache.",
```

File: public/static/locales/en/data.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "cancel": "Cancel",
     "communityData": "Community Data",
     "copyPath": "Copy Path",
-    "copyLinks": "Copy Public Link(s)",
+    "copyLinks": "Copy",
```
Comment (psarando) [inline public/static/locales/en/data.json:21]: I'd be tempted to leave this alone, but change the button label to use `common:copy` instead 🤔

File: src/components/data/PublicLinks.js [hunk 3]
```diff
@@ -186,47 +202,80 @@ function PublicLinks(props) {
...
-                            className={classes.toolbarButtons}
-                            onClick={() => setDownload(true)}
-                            startIcon={<CloudDownload />}
-                            id={build(baseId, ids.DOWNLOAD_LINKS_BTN)}
-                        >
-                            <Hidden xsDown>{t("download")}</Hidden>
-                        </Button>
-                    </Tooltip>
-                </Toolbar>
+                <TextareaAutosize
```
Comment (psarando) [inline src/components/data/PublicLinks.js:205]: 👍

File: src/components/data/PublicLinks.js [hunk 4]
```diff
@@ -186,47 +202,80 @@ function PublicLinks(props) {
...
+                    rowsMin={paths?.length + 1 || 3}
```
Comment (psarando) [inline src/components/data/PublicLinks.js:206]: Seems like adding logic based on the contents to the `rowsMin` prop in `TextareaAutosize` defeats its purpose?
```suggestion
                    rowsMin={3}
```

File: src/server/configuration.js [hunk 5]
```diff
@@ -284,3 +285,10 @@ export const uatURL = config.get("services.unified_astronomy_thesaurus.base");
  * The base URL for the User Portal's API
  */
 export const userPortalURL = config.get("services.user_portal_api.base");
+
+/**
+ *
+ * The base ULR for Data Store anon files
```
Comment (psarando) [inline src/server/configuration.js:291]:
```suggestion
 * The base URL for Data Store anon files
```

File: src/serviceFacades/filesystem.js [hunk 6]
```diff
@@ -272,6 +272,25 @@ export const getPublicLinks = (key, paths) => {
...
+                    callApi({
+                        endpoint: `/api/filesystem/refresh-cache?path=${path}`,
+                        method: "GET",
+                    })
+                )
+            )
+                .then((values) => resolve(values))
+                .catch((error) => reject(error))
+        )
+    );
```
Comment (psarando) [inline src/serviceFacades/filesystem.js:291]: Pretty sure `Promise.all` doesn't need to be wrapped in another Promise.
```suggestion
            Promise.all(
                paths.map((path) =>
                    callApi({
                        endpoint: `/api/filesystem/refresh-cache?path=${path}`,
                        method: "GET",
                    })
                )
            );
```

**Review by psarando — APPROVED** [review]
LGTM 👍 

I only had a couple of minor suggestions.

File: src/serviceFacades/filesystem.js (hunk 6 above)
Comment (psarando) [reply src/serviceFacades/filesystem.js:291]: Ah, yep, found another example:
https://github.com/cyverse-de/sonora/blob/3071bf50a9a46b3dab7d383cc47a2150c1caba34/src/serviceFacades/tools.js#L153-L159

Comment (sri2k1us) [reply src/serviceFacades/filesystem.js:291]: Hmm. I was looking for example when I wrote this and found this way and then https://github.com/cyverse-de/sonora/blob/3071bf50a9a46b3dab7d383cc47a2150c1caba34/src/serviceFacades/analyses.js#L143 😕

**Review by ash3rz — APPROVED** [review]
LGTM :shipit:

**Comment by sri2k1us** [comment]
Thanks for the reviews.

**Pull request merged**